src/libtiled/mapreader.cpp

```diff
@@ -297,6 +297,8 @@ std::unique_ptr<Map> MapReaderPrivate::readMap()
             mMap->addLayer(std::move(layer));
         else if (xml.name() == QLatin1String("properties"))
             mMap->mergeProperties(readProperties());
+        else if (xml.name() == QLatin1String("components"))
+            mMap->mergeComponents(readComponents());
```

Comment:
Since we now support properties and components everywhere, it probably makes sense to introduce a helper function like `readCommonObjectElement`, so we can change this to:
```c++
else if (!readCommonObjectElement(object))
    readUnknownElement();
```

Reply:
I think the existing solution is much more readable. There isn't a lot of bilerplate while `readCommonObjectElement` is kind of ambiguous.